### ISSUE TITLE
_native_batch_norm_legit doesn't bump version counter of buffers

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#120622]
The op's schema looks like the following:
<img width="730" alt="image" src="https://github.com/pytorch/pytorch/assets/5652049/9e4d3c8b-869e-445b-98c4-360ec63a7b0e">
repro:
```
x = torch.randn(2, 3, 4, 4)
w = torch.randn(3)
running_mean = torch.zeros(3)
running_var = torch.zeros(3)
print(running_mean._version)
torch.ops.aten._native_batch_norm_legit(x, w, None, running_mean, running_var, True, 0.1, 0.001)
print(running_mean._version)  # 0
```

**Comment by zou3519** [comment]
hi-pri for potential silent incorrectness, but not sure how important this is because no one has run into it.

**Comment by bdhirsh** [comment]
cc @andrewor14 - we should make sure that the new batch norm op doesn't have this problem
